### PR TITLE
PMM-1797 Fix CloudWatch data source

### PR DESCRIPTION
**Pull request by AlekSi**

### CODE REVIEW

File: import-dashboards.py [hunk 1]
```diff
@@ -105,7 +148,7 @@ def add_datasources():
             sys.exit(-1)
 
     if 'CloudWatch' not in ds:
-        data = json.dumps({'name': 'CloudWatch', 'type': 'cloudwatch', 'jsonData': '{"defaultRegion":"us-east-1"}', 'access': 'proxy', 'isDefault': False})
```
Comment (AlekSi) [inline import-dashboards.py]: The root cause is there: `jsonData` should be a nested JSON object, not encoded JSON string. Grafana API does not validate it and stores as-is, leading to JS error on the edit page.

File: import-dashboards.py [hunk 2]
```diff
@@ -92,6 +108,33 @@ def delete_api_key(upgrade):
...
+def fix_cloudwatch_datasource():
+    """
+    Replaces incorrect CloudWatch datasource stored as JSON string with correct JSON object.
+    """
+
+    con = sqlite3.connect(GRAFANA_DB_DIR + '/grafana.db')
+    cur = con.cursor()
+
+    found = False
+    cur.execute("SELECT id, json_data FROM data_source WHERE name = 'CloudWatch'")
```
Comment (delgod) [inline import-dashboards.py]: likely, it is impossible to create two datasources with the same name

Comment (AlekSi) [reply import-dashboards.py]: But it may be removed by a user.

File: import-dashboards.py [hunk 3]
```diff
@@ -92,6 +108,33 @@ def delete_api_key(upgrade):
...
+
+        old = None
+        try:
+            old = json.loads(row[1])
+        except:
+            pass
+
+        if not isinstance(old, dict):
+            new = {'authType': 'keys'}
+            cur.execute("UPDATE data_source SET json_data = ? WHERE id = ?", (json.dumps(new), row[0]))
```
Comment (delgod) [inline import-dashboards.py]: is documented credentials file way still working after this modification?
https://www.percona.com/doc/percona-monitoring-and-management/amazon-rds.html#monitoring-amazon-rds-os-metrics

Comment (AlekSi) [reply import-dashboards.py]: It should, but it is definitely should be tested by QA.

**Comment by peterschwaller** [comment]
touch

**Pull request merged**